grr/server/grr_response_server/gui/ui/store/client_page_facade.ts

```diff
@@ -355,9 +356,18 @@ export class ClientPageStore extends ComponentStore<ClientPageState> {
   /** An effect querying results of a given flow. */
   private readonly queryFlowResultsImpl = this.effect<FlowResultsQuery>(
       obs$ => obs$.pipe(
-          takeUntil(this.selectedClientIdChanged$),
-          withLatestFrom(this.selectedClientId$),
-          exhaustMap(([query, clientId]) => {
+        takeUntil(this.selectedClientIdChanged$),
+        withLatestFrom(this.selectedClientId$),
+        // Below we are grouping the requests by flowId, tag and type, and
+        // for each group we use a queuedExhaustMap with queue size 1 to send http request
+        // for results.
+        // TODO: Think about how we should split the code so it's cleaner
+        // TODO: Add tests to ensure the intended functionality
+        groupBy(([query, clientId]) => {
+          return [query.flowId, query.withTag, query.withType].join('-');
+        }),
+        mergeMap(group$ => group$.pipe(
+          queuedExhaustMap(([query, clientId]) => {
```

Comment:
Try different approaches for making this cleaner. But it's also not-that-awful this way. We could leave it like this for now, and make it better as part of the general refactoring of `client_page_facade.ts`.